### ISSUE TITLE
domain-specific TARGET_DOMAIN?

### ISSUE THREAD

**Issue by jo-me** [tiredofit/docker-traefik-cloudflare-companion#28]
I have two zones in Cloudflare, so 2 domains and separate DNS settings.
In traefik I specify dedicated subdomains for each domain.

cfcompanion's config allows to specifiy several domains DOMAIN1, DOMAIN2...
However, the TARGET_DOMAIN seems to be used for all domains specified.
All CNAME records are created in that TARGET_DOMAIN.

So in my domain1.com DNS settings I get
```
CNAME    mysubdomain.domain2.com       domain1.com
CNAME    othersubdomain                domain1.com
```

Instead of one CNAME in domain1.com and the other in domain2.com
 
Bug ?  Or am I not getting something?

**Comment by jo-me** [comment]
My workaround is to create two containers - one for each domain/zone

**Comment by tiredofit** [comment]
I'd have to think about this but it would likely involve a massive rewrite of the code. I myself have a similar solution with 9 zone's. How I solve it is by creating a A record to the hostname of the server itself.

**Comment by mbjurstrom** [comment]
I think I managed to implement this in https://github.com/tiredofit/docker-traefik-cloudflare-companion/pull/30. Can someone else test it as well and see that I didn't make a stupid mistake. 

It seem to work fine for me at least, using it right now together with the changes from https://github.com/tiredofit/docker-traefik-cloudflare-companion/pull/15.

**Comment by mbjurstrom** [comment]
This should be fixed now that #30 is merged

**Comment by tiredofit** [comment]
Tagged as `tiredofit/traefik-cloudflare-companion:6.2.0`  - Many thanks @mbjurstrom
